File: Python algo/strings/5-palindrome.py
import re

class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = re.sub(r'[^a-zA-Z0-9]*', '', s).lower()

        for i in range(len(s)):
            if s[i] != s[len(s)-1-i]:
                return False

        return True

s = "A man, a plan, a canal: Panama"
sol = Solution()
print(sol.isPalindrome(s))

File: js algo/strings/5-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
// best solution : use pointers to compare each value without editing the string
 var isPalindrome = function(s) {
    purifiedS = s.replace(/[^a-zA-Z0-9]*/g, '').toLowerCase()

    for (let i = 0; i < purifiedS.length; i++) {
        
        if(purifiedS[i] !== purifiedS[purifiedS.length-1-i])
            return false
    }

    return true
};

console.log(isPalindrome('A man, a plan, a canal: Panama'))